### PR TITLE
Add newsfeed methods to saved search model

### PR DESCRIPTION
**Pull request by ryanhertz**
I added a model for NewsFeeds and some related methods to SavedSearch.

### CODE REVIEW

File: lib/spark_api/models/newsfeed.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module SparkApi
+  module Models
+    class Newsfeed < Base 
+      
+      extend Finders
+
+      self.element_name = 'newsfeeds'
+
+      def update!(arguments={})
+        connection.put( "/newsfeeds/#{self.Id}", arguments ).first
```
Comment (bhornseth) [inline lib/spark_api/models/newsfeed.rb]: One issue here: the instance attributes will get out of sync with what's actually persisted to the API. If you pull in the `Savable` concern, that'll provide the `update!` method as well as handle the dirty tracking. The only difference is the usage would look like:

```ruby
newsfeed = Newsfeed.find('1234')
newsfeed.Active = false
newsfeed.save!
```

File: lib/spark_api/models/saved_search.rb [hunk 2]
```diff
@@ -6,12 +6,25 @@ class SavedSearch < Base
...
+
       def self.provided()
         Class.new(self).tap do |provided|
           provided.element_name = '/savedsearches'
           provided.prefix = '/provided'
+          def is_provided_search?
+            true
+          end
+          def newsfeeds
+            []
```
Comment (bhornseth) [inline lib/spark_api/models/saved_search.rb]: small discrepancy: the default initializer sets `@newsfeeds` to nil, here it's an empty array. Should those be the same?

File: lib/spark_api/models/saved_search.rb [hunk 3]
```diff
@@ -49,6 +62,65 @@ def contacts
         end
       end
 
+      def listings(args = {})
+        arguments = {:_filter => self.Filter}.merge(args)
+        @listings ||= Listing.collect(connection.get("/listings", arguments))
+      end
+
+      def newsfeeds
+        if @newsfeeds.nil?
```
Comment (bhornseth) [inline lib/spark_api/models/saved_search.rb]: ok, the nil / empty array thing makes sense now

**Comment by bhornseth** [comment]
few comments, but otherwise :+1: on putting these in the models

File: lib/spark_api/models/newsfeed.rb (hunk 1 above)
Comment (ryanhertz) [inline lib/spark_api/models/newsfeed.rb]: Thanks, I noticed the `Savable` concern, but I was using it incorrectly. I meant to ask about it in the pr.

File: lib/spark_api/models/saved_search.rb (hunk 2 above)
Comment (ryanhertz) [inline lib/spark_api/models/saved_search.rb]: I was having some problems with the tests and this provided class modifier method. I thought about creating a subclass, now that there are two separate attributes and two new methods. I set the provided newsfeeds to `[]` mainly to get the tests working. :disappointed:

**Pull request merged**